### PR TITLE
Simpler name and directory check

### PR DESCRIPTION
**Pull request by Nirma**
I thought this may be a more straight forward and compact way to go about these checks. 

Hope it helps!

### CODE REVIEW

File: Source/CarthageKit/Git.swift [hunk 1]
```diff
@@ -247,23 +247,17 @@ public func cloneSubmoduleInWorkingDirectory(submodule: Submodule, workingDirect
...
-				if name != ".git" {
-					return .empty
-				}
-			} else {
-				return .empty
-			}
-
+            if (name as? String) != ".git" {
+                return .empty
+            }
```
Comment (mdiep) [inline Source/CarthageKit/Git.swift]: It looks like the indentation is off here and below. Can you switch both areas back to tabs?

Comment (Nirma) [inline Source/CarthageKit/Git.swift]: @mdiep Sorry about that! My xvim settings got in the way here, I will be more careful next time!

Comment (mdiep) [inline Source/CarthageKit/Git.swift]: No worries! It's easy to miss. :smile:

**Comment by mdiep** [comment]
Thanks for the pull request!

**Pull request merged**